### PR TITLE
feat: functions operating on all elements in the time series

### PR DESCRIPTION
**Pull request by gunnarstaff**
Added statistical calculations like time weighted mean, time weighted standard deviation and min/max operators that work over all the values in the time series and not element wise

## Description
Only adding new functions.

## Motivation and Context
The new functionality allows you to solve problems that previously have been very hard to solve. Now we can create time series based on statistical properties from the entire time series. We can create scatter plots that are overlayed in the time series plot,+++

## How Has This Been Tested?
All four functions have been tested on a real time series taken from the demo tenant.
Tested on Windows 11 with python 3.10.7

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

## Contributor Checklist:

- [x] My code follows the code style of this project.
- [ ] I have added an example of my new feature and included it in the documentation.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** [document](https://cognitedata.github.io/indsl/contribute.html).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ x My Pull Request name follows the [naming convention](https://www.conventionalcommits.org/en/v1.0.0/) `fix: <description>`, `feat: <description>`, etc.

## Reviewer Checklist for Charts compliant functions:

- [x] The docstrings of the new function follow the contributing [guidelines](https://cognitedata.github.io/indsl/contribute.html).
- [x] The new function is professionally [documented](https://journal.emwa.org/writing-matters/whats-your-problem-a-practical-approach-to-scientific-document-design/article/2110/2047480612z2e00000000042.pdf)
- [x] The new function and associated scripts are covered by one or more unit tests and code coverage did not decrease.
- [ ] The new function is accompanied by an example and it is included in the Gallery of Charts.
- [ ] The new function is reviewed in Chromatic. Access the storybook build results url and comment, approve or deny.
- [x] All function inputs, arguments, and outputs have a supported data type and have human readable names.
- [x] No code language is included in the description of the function or parameters (e.g use "polynomial order" instead of "poly_order")

### CODE REVIEW

**Review by VebjornG — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -151,3 +151,5 @@ localdev-*
 # OS generated
 .DS_Store
 .DS_Store?
+/docs-source/source/auto_examples
+/docs-source/source/auto_examples
```
Comment (neringaalt) [inline .gitignore:155]: You should not commit .`gitignore` in this PR. 
auto_examples will be used when we release documentation, so we should not ignore it in the main branch, but you should ignore it in your PR :)

File: indsl/ts_utils/array_operations.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    Args:
+        data: Time series.
+
+    Returns:
+        pandas.Series: Time weighted mean.
+    """
+    from scipy.integrate import trapezoid
+
+    n = len(data)
```
Comment (neringaalt) [inline indsl/ts_utils/array_operations.py:24]: you can use `validate_series_is_not_empty` function that is implemented in `validations.py` so you will avoid repeating the code

File: indsl/ts_utils/array_operations.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+    Args:
+        data: Time series.
+
+    Returns:
+        pandas.Series: Time weighted std.
+    """
+    from scipy.integrate import trapezoid
+
+    n = len(data)
```
Comment (neringaalt) [inline indsl/ts_utils/array_operations.py:55]: `validate_series_is_not_empty` could be use here as well and any other place where you check for `n==0`

File: tests/ts_utils/test_array_operations.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Cognite AS
```
Comment (neringaalt) [inline tests/ts_utils/test_array_operations.py:1]: not a big deal, but maybe 2024 :D

File: .gitignore (hunk 1 above)
Comment (gunnarstaff) [reply .gitignore]: Committed by accident. Should be corrected now.

File: tests/ts_utils/test_array_operations.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import pandas as pd
+import math
+import pytest
+
+from indsl.ts_utils import time_weighted_mean, time_weighted_std, timeseries_min, timeseries_max
+
+
+@pytest.mark.core
+def test_time_weighted_mean():
+    WHP_series = pd.read_pickle("./tests/ts_utils/pd_series_WHP.pkl")
```
[the remark this answers is not in the document]
Comment (gunnarstaff) [reply tests/ts_utils/test_array_operations.py:12]: It is sanitized data. Is that ok?

File: tests/ts_utils/test_array_operations.py (hunk 4 above)
Comment (gunnarstaff) [reply tests/ts_utils/test_array_operations.py]: Corrected

**Review by neringaalt — APPROVED** [review]

**Review by VebjornG — APPROVED** [review]

**Pull request merged**